test/test-packages.py

```diff
@@ -116,20 +118,45 @@ def get_test_set():
 
 
 process_work_dir = ''
+
+
 def do_test_initializer():
     global process_work_dir
     process_work_dir = "work_pid_%d" % os.getpid()
     os.mkdir(process_work_dir)
     subprocess.run(f'cp container-* {process_work_dir}/', shell=True)
 
+
 def do_test_lambda(x):
     return do_test(*x)
+
+
+def process_pip_report(report, package):
+    result = None
+    try:
+        f = open(report)
```

Comment:
It's better to use the `with open(report) as f:` style so that the file gets closed even if there's an exception.